### PR TITLE
Fix radio other grouping

### PR DESCRIPTION
**Pull request by drusepth**
This PR includes the following changes:

- Binds our `other` radio boxes to the same model property as other radio selections on the field, ensuring the radio functionality of "if I click this one, it deselects all others" (c.f. https://github.com/dobtco/screendoor-v2/issues/4557)

- Adds an example radio field with a true `include_other_option` to our examples/index.html kitchen sink, for demonstrating the functionality of showing an "other" input when the other option is selected (and hidden again when a non-other option is selected):

2017-07-18-201119_471x99_scrot

2017-07-18-201134_374x98_scrot

- Updates the tests to ensure the field serializes correctly when blank and when the other option is selected.

- Includes the latest dist/ build. 

It still needs to be tested from within Screendoor, but I think we can follow a similar update/test process to the approach we took in #142 (i.e. release new version upon PR merge, then test Screendoor on staging with it).

### CODE REVIEW

**Comment by vlymar** [comment]
Looks like you decided not to update rivets?

**Comment by drusepth** [comment]
Yeah, when I looked at upgrading Rivets it actually looked pretty straightforward and didn't seem to break anything, but since it wasn't required for this fix I figured we'd exclude it from the changes, get this fixed, and upgrade as a standalone change whenever we decided we wanted to.

**Comment by drusepth** [comment]
@vlymar This is ready for review at your convenience.

File: src/vendor_config.coffee [hunk 1]
```diff
@@ -18,6 +18,12 @@ rivets.binders.input =
   unbind: (el) ->
     $(el).unbind("#{rivets.inputEvent}.rivets")
 
+rivets.formatters.eq = (value, checkAgainst) ->
+  if value.constructor == Array
+    value[0] == checkAgainst
```
Comment (vlymar) [inline src/vendor_config.coffee]: why does this only check the first value?

File: src/templates/partials/options_field.eco [hunk 2]
```diff
@@ -10,10 +10,10 @@
 <% if @model.get('include_other_option'): %>
   <div class='fr_option fr_other_option'>
     <label class='control'>
-      <input type='<%= fieldType %>' data-rv-checked='model.value.other_checked' />
+      <input type='<%= fieldType %>' data-rv-checkedarray="model.value.checked" value="<%= FormRenderer.t.other %>" />
```
Comment (vlymar) [inline src/templates/partials/options_field.eco]: can you explain the changes in this file?

**Comment by drusepth** [comment]
@vlymar I updated the formatter to look at any value in the selected options, rather than just the first one, which lets the other textbox show up for checkbox fields that allow other as well (which was already working, but would have broken with this change -- good catch). I also added a test for checkboxes with an other option. 

Ready for you to review again.

This does mean we still have the minor bug where:

* If a user turns on the "Include an Other option" _and_ adds an option for "Other" manually,
* Then clicking either will select both and show the Other reason text box on the latter, e.g.

2017-07-24-185754_716x373_scrot

I think that's a more minimal bug that we can tackle separately, if you think this fixes the more pressing bug.

**Comment by vlymar** [comment]
LGTM 👍

**Pull request merged**